x-pack/legacy/plugins/uptime/public/components/functional/monitor_list/monitor_list_drawer/monitor_status_list.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext } from 'react';
+import { get } from 'lodash';
+import { EuiHealth, EuiSpacer, EuiCallOut } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { Check } from '../../../../../common/graphql/types';
+import { UptimeSettingsContext } from '../../../../contexts';
+import { LocationLink } from './location_link';
+
+interface MonitorStatusListProps {
+  /**
+   * Recent List of checks performed on monitor
+   */
+  checks: Check[];
+}
+
+export const MonitorStatusList = ({ checks }: MonitorStatusListProps) => {
+  const {
+    colors: { success, danger },
+  } = useContext(UptimeSettingsContext);
+
+  const upChecks: Set<string> = new Set();
+  const downChecks: Set<string> = new Set();
+
+  const UN_NAMED_LOCATION = 'unnamed-location';
+
+  checks.forEach((check: Check) => {
+    let location = get<string>(check, 'observer.geo.name', UN_NAMED_LOCATION);
+
+    if (location === null) {
+      location = UN_NAMED_LOCATION;
+    }
+
+    if (check.monitor.status === 'up') {
+      upChecks.add(location);
+    } else if (check.monitor.status === 'down') {
+      downChecks.add(location);
+    }
+  });
+
+  // if monitor is down in one dns, it will be considered down so removing it from up list
+
+  const absUpChecks: Set<string> = new Set([...upChecks].filter(item => !downChecks.has(item)));
+
+  const displayMonitorStatus = (checksList: Set<string>, color: string, titleTxt: string) => {
+    let checkListArray = [...checksList];
+    // If un-named location exists, move it to end
+    if (checksList.has(UN_NAMED_LOCATION)) {
+      checkListArray = checkListArray.filter(item => item !== UN_NAMED_LOCATION);
+      checkListArray.push(UN_NAMED_LOCATION);
+    }
+
+    return (
+      <>
+        <EuiHealth color={color}>
+          {titleTxt} in {checkListArray.map((location, index) => (index ? ', ' : '') + location)}
+        </EuiHealth>
+        <EuiSpacer size="s" />
+      </>
+    );
+  };
```

Comment:
It might be cleaner to extract this function to its own component and invoke it in the JSX block below; we could have it return `null` when the `Set` is empty and clean up the markup in this file a bit.

Subjective though so if you disagree it's no reason to hold up the patch.

Reply:
also done !!